test/parallel/test-cluster-worker-wait-server-close.js

```diff
@@ -22,10 +22,12 @@ if (cluster.isWorker) {
   worker.once('listening', function() {
     net.createConnection(common.PORT, common.localhostIPv4, function() {
       var socket = this;
-      worker.disconnect();
-      setTimeout(function() {
-        socket.write('.');
-        connectionDone = true;
+      setTimeout(function(){
```

Comment:
Please, see this Gist: https://gist.github.com/Olegas/a6a03c65ed533e89a9a9
Trying to explain the best I can.